Add match operator in ColumnExpression for FTS5.

GetToSet opened the pull request:
<!-- Please describe your pull request here. -->

### Pull Request Checklist

<!--
Please verify that your pull request checks those boxes:
-->

- [x] This pull request is submitted against the `development` branch.
- [x] Inline documentation has been updated.
- [x] README.md or another dedicated guide has been updated.
- [x] Changes are tested.

I'm using GRDB.swift with FTS5 support in my own project and found that in FTS5 query interface, there is no `match` operator for column expressions, which is found in FTS3.

This PR add this operator expression and related tests, making it aligned with FTS3 query interface API.

groue commented:
Thank you @GetToSet, [FTS5 column filters](https://www.sqlite.org/fts5.html#fts5_column_filters) were lacking 👍

groue commented:
I removed the support for `column.match(nil)`, since SQLite throws a syntax error for `MATCH NULL`.

I also updated the documentation.

GetToSet commented:
> I removed the support for `column.match(nil)`, since SQLite throws a syntax error for `MATCH NULL`.
> 
> I also updated the documentation.

`column.match(nil)` also presents in FTS3 (FTS3+QueryInterface.swift), should it also be disabled?

groue commented:
No, because FTS3 accepts NULL. An little harmless SQLite inconsistency ;-)

GetToSet commented:
> No, because FTS3 accepts NULL. An little harmless SQLite inconsistency ;-)



> No, because FTS3 accepts NULL. An little harmless SQLite inconsistency ;-)

🤔 I've never dived such detailed into SQLite grammar, BTW, thanks for instant response for this PR.

groue commented:
I just ran the tests, and saw that nil matching was throwing an error.

Since you are contributing many FTS5 enhancements to GRDB, please let me share a tip. When you open the GRDB workspace, you can run tests with FTS5 enabled by modifying `GRDBDeploymentTarget.xcconfig`:

```
// Add this line to GRDBDeploymentTarget.xcconfig
OTHER_SWIFT_FLAGS = $(inherited) -D SQLITE_ENABLE_FTS5
```

You can then select the GRDBOSX schema, and run FTS5 tests.

Do not commit this change! The raw project must remain FTS5-free so can users can embed it in their apps that target older operating systems.

groue commented:
Shipped in v5.10.0
